# Output

## Introduction

This document describes the output produced by the pipeline. Most of the plots are taken from the MultiQC report, which summarises results at the end of the pipeline.

The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.

<!-- TODO nf-core: Write this documentation describing your workflow's output -->

## Pipeline overview

The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:

- [FastQC](#fastqc) - Illumina read QC
- [NanoPlot](#nanoplot) - ONT read QC
- [Filtlong](#filtlong) - Quality filtering long reads
- [Ratatosk](#ratatosk) - Hybrid error correction of long reads
- [Flye](#flye) - *De novo* assembler
- [Racon](#racon) - Assembly polishing using long-reads
- [Pilon](#pilon) - Assembly polishing using short-reads
- [Purge_Dups](#purge_dups) - Purge haplotigs and overlaps
- [RagTag](#ragtag) - Contig correction and scaffolding
- [TGS-GapCloser](#tgs-gapcloser) - Close gaps
- [QUAST](#quast) - Quality assessment of genome assembly
- [Merqury](#merqury) - k-mer based assembly evaluation
- [BUSCO](#busco) - Single-Copy Orthologs based assessment
- [MultiQC](#multiqc) - Aggregate QC results
- [Pipeline general information](#pipeline-general-information) - Report metrics generated during the workflow execution

### FastQC

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/fastqc/`
  - `*_fastqc.html`: FastQC report containing quality metrics.
  - `*_fastqc.zip`: Zip archive containing the FastQC report, tab-delimited data file and plot images.

</details>

### NanoPlot

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/nanoplot_*/`
  - `*.html`: Nanoplot HTML reports.
  - `*.png`: Plots generated by NanoPlot.
  - `*.txt`: Stats from NanoPlot.
  - `*.log`: Log file of NanoPlot run.

</details>

### Filtlong

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/*.fastq.gz`: Filtered (compressed) FASTQ file.

</details>

### Ratatosk

<details markdown="1">
<summary>Output files</summary>

</details>

### Flye

<details markdown="1">
<summary>Output files</summary>

</details>

### Racon

<details markdown="1">
<summary>Output files</summary>

</details>

### Pilon

<details markdown="1">
<summary>Output files</summary>

</details>

### Purge_Dups

<details markdown="1">
<summary>Output files</summary>

</details>

### RagTag

<details markdown="1">
<summary>Output files</summary>

</details>

### TGS-GapCloser

<details markdown="1">
<summary>Output files</summary>

</details>

### QUAST

<details markdown="1">
<summary>Output files</summary>

</details>

### Merqury

<details markdown="1">
<summary>Output files</summary>

</details>

### BUSCO

<details markdown="1">
<summary>Output files</summary>

</details>

### MultiQC

<details markdown="1">
<summary>Output files</summary>

- `multiqc/`
  - `multiqc_report.html`: a standalone HTML file that can be viewed in your web browser.
  - `multiqc_data/`: directory containing parsed statistics from the different tools used in the pipeline.
  - `multiqc_plots/`: directory containing static images from the report in various formats.

</details>

### Pipeline general information

<details markdown="1">
<summary>Output files</summary>

- `pipeline_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.dot`/`pipeline_dag.svg`.
  - Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.yml`. The `pipeline_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the pipeline.
  - Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.
  - Parameters used by the pipeline run: `params.json`.

</details>
