# Output

## Introduction

This document describes the output produced by the pipeline. Most of the plots are taken from the MultiQC report, which summarises results at the end of the pipeline.

The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.

## Pipeline overview

The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:

- [FastQC](#fastqc) - Illumina read QC
- [NanoPlot](#nanoplot) - ONT read QC
- [Filtlong](#filtlong) - Quality filtering long reads
- [Ratatosk](#ratatosk) - Hybrid error correction of long reads
- [Flye](#flye) - *De novo* assembler
- [Racon](#racon) - Assembly polishing using long-reads
- [Pilon](#pilon) - Assembly polishing using short-reads
- [Purge_Dups](#purge_dups) - Purge haplotigs and overlaps
- [RagTag](#ragtag) - Contig correction and scaffolding
- [TGS-GapCloser](#tgs-gapcloser) - Close gaps
- [QUAST](#quast) - Quality assessment of genome assembly
- [Merqury](#merqury) - k-mer based assembly evaluation
- [BUSCO](#busco) - Single-Copy Orthologs based assessment
- [MultiQC](#multiqc) - Aggregate QC results
- [Pipeline general information](#pipeline-general-information) - Report metrics generated during the workflow execution

### FastQC

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/qcs/fastqc/`
  - `*_fastqc.html`: FastQC report containing quality metrics.
  - `*_fastqc.zip`: Zip archive containing the FastQC report, tab-delimited data file and plot images.

</details>

### NanoPlot

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/qcs/nanoplot_*/`
  - `*.html`: Nanoplot HTML reports.
  - `*.png`: Plots generated by NanoPlot.
  - `*.txt`: Stats from NanoPlot.
  - `*.log`: Log file of NanoPlot run.

</details>

### Filtlong

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/*.filtered.fastq.gz`: Filtered (compressed) FASTQ file.

</details>

### Ratatosk

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/*.corrected.fastq.gz`: Corrected (compressed) FASTQ file.

</details>

### Flye

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/assembly/`
  - `*.fasta.gz`: Assembled FASTA file.
  - `*.gfa.gz`: Repeat graph in gfa format.
  - `*.gv.gz`: Repeat graph in gv format.
  - `*.txt`: Extra information and statistics about resulting contigs.
  - `*.log`: Flye log file
  - `*.json`: Flye parameters

</details>

### Racon

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/polishing/`
  - `*_assembly_consensus.fasta.gz`: Improved genome assembly.

</details>

### Pilon

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/polishing/`
  - `*.pilon.fasta.gz`: Improved genome assembly with short-reads.

</details>

### Purge_Dups

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/curation/purgedups/`
  - `*.calcuts.log`: Log file.
  - `*.cutoffs`: Cutoff file.
  - `*.dups.bed`: A bed file of sequences purged of false duplications.
  - `*.hap.fa`: Fasta file containing purged haplotigs.
  - `*.paf`: PAF alignment file.
  - `*.PB.base.cov`: Base coverage file.
  - `*.PB.stat`: Statistic file.
  - `*.purged.fa`: Fasta file purged of duplicated haplotigs.
  - `*.purge_dups.log`: A log of the tool output
  - `*.split.fasta.gz`:
  - `*.split.paf`: Fasta split by Nâ€™s
</details>

### RagTag

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/curation/scaffolds/`
  - `*.corrected.fasta`: The corrected query assembly in FASTA format.
  - `*.scaffold.fasta`: The scaffolds in FASTA format, defined by the ordering and orientations of ragtag.scaffold.agp.
  - `*.scaffold.agp`: The ordering and orientations of query sequences in AGP format.
  - `*.scaffold.stats`: Summary statistics for the scaffolding process.

</details>

### TGS-GapCloser

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/curation/gapfilling/`
  - `*.fasta`: The final assembly after gap filling.
  - `*.gap_fill_details`: Details about how the final assembly was assembled.

</details>

### QUAST

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/qcs/quast/`
  - `<SAMPLE>/`: Directory containing the results of the QUAST analysis.
  - `*.tsv`: TSV file.
  - `*_misassemblies.tsv`: Report containing misassemblies, only when a reference fasta is provided.
  - `*_unaligned.tsv`: Report containing unaligned contigs, only when a reference fasta is provided.

</details>

### Merqury

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/qcs/merqury/`
  - `*.unionsum.meryl/`: A Meryl k-mer database that is the union sum of the input databases
  - `*.meryl/`: A Meryl k-mer database
  - `*.completeness.stats`: Assembly statistics file.
  - `*.dist_only.hist`: Histogram.
  - `*_only.bed`: The positions of the k-mers found only in an assembly for further investigation in .bed.
  - `*_only.wig`: The positions of the k-mers found only in an assembly for further investigation in .wig.
  - `*.qv`: Assembly consensus quality estimation
  - `*.spectra-asm.fl.png`: Unstacked assembly spectra filled plot in PNG format.
  - `*.spectra-asm.hist`: Assembly spectra histogram.
  - `*.spectra-asm.ln.png`: Unstacked assembly spectra line plot in PNG format.
  - `*.spectra-asm.st.png`: Stacked assembly spectra line plot in PNG format
  - `*.unionsum.hist.ploidy`: Ploidy estimate from read k-mer database

</details>

### BUSCO

<details markdown="1">
<summary>Output files</summary>

- `<SAMPLE>/qcs/busco/`
  - `*-busco/`: BUSCO lineage specific output.
  - `*-busco.batch_summary.txt`: Summary of all sequence files analyzed.
  - `short_summary.*.json`: Short Busco summary in JSON format.
  - `short_summary.*.txt`: Short Busco summary in plain text format.

</details>

### MultiQC

<details markdown="1">
<summary>Output files</summary>

- `multiqc/`
  - `multiqc_report.html`: a standalone HTML file that can be viewed in your web browser.
  - `multiqc_data/`: directory containing parsed statistics from the different tools used in the pipeline.
  - `multiqc_plots/`: directory containing static images from the report in various formats.

</details>

### Pipeline general information

<details markdown="1">
<summary>Output files</summary>

- `pipeline_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.dot`/`pipeline_dag.svg`.
  - Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.yml`. The `pipeline_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the pipeline.
  - Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.
  - Parameters used by the pipeline run: `params.json`.

</details>
